### PR TITLE
Refactor the two reservation unit state helpers into lookup properties

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Replaces `ReservationUnitStateHelper` and `ReservationUnitReservationStateHelper` with lookup properties
- Renames Reservation Unit `state` to `publishing_state` (required due to conflicts when using lookup properties, but also a better name)
- Use new states optimally in GraphQL API
- Refactor tests to support new states
- Remove lookup properties from audit logs
- Convert state enums to TestChoices
- Add some typing infor for refactoring purposes

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- Requires frontend to support new filters, and renaming of reservation unit `state` to `publishing_state`. Should be a quick fix.

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3404][TILA]

[TILA]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3404

[TILA-3404]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3404?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reservation_units/models/reservation_unit.py [hunk 1]
```diff
@@ -268,23 +272,228 @@ class Meta:
...
+                    & (
+                        # Was always published.
+                        models.Q(publish_begins__isnull=True)
+                        # Was published in the past.
+                        | models.Q(publish_begins__lte=Now())
+                    )
+                ),
+                then=models.Value(ReservationUnitPublishingState.SCHEDULED_HIDING.value),
+            ),
+            # Reservation Unit has not been published, but has is going to be published,
```
Comment (vergama) [inline reservation_units/models/reservation_unit.py:384]: 'has is'

File: reservation_units/models/reservation_unit.py [hunk 2]
```diff
@@ -268,23 +272,228 @@ class Meta:
...
+            #  OR it was reservable in the past (but not anymore),
+            #  OR reservation period is zero-length.
+            models.When(
+                (
+                    # No active pricing.
+                    L(active_pricing_type=None)
+                    | (
+                        # Paid pricing with no payment product.
+                        L(active_pricing_type=models.Value(PricingType.PAID.value))
+                        & models.Q(payment_product__isnull=True)
```
Comment (vergama) [inline reservation_units/models/reservation_unit.py:448]: is the check for.MOCK_VERKKOKAUPPA_API_ENABLED no longer necessary?

Comment (matti-lamppu) [reply reservation_units/models/reservation_unit.py:448]: It is necessary, I just forgot to add it back in. Will fix on Monday.

Comment (matti-lamppu) [reply reservation_units/models/reservation_unit.py:449]: Done.

**Review by ranta — APPROVED** [review]

**Pull request closed without merging**